src/data/hicdex.js

```diff
@@ -207,3 +242,37 @@ export async function fetchGraphQL(operationsDoc, operationName, variables) {
 
   return await result.json()
 }
+
+export async function getObjktsByShare(addresses, min_shares) {
+  const { errors, data } = await fetchGraphQL(
+    getCollabTokensForAddressesByShare,
+    'GetCollabTokens',
+    { addresses }
+  )
+
+  if (errors) {
+    console.error(errors)
+  }
+  let objkts = []
+  const contracts_info = data.hic_et_nunc_splitcontract
+  for (const i in contracts_info) {
+    const contract = contracts_info[i].contract
+    const total = contract.shares[0].total_shares
+    const tokens = _.filter(
+      contract.shares[0].shareholder.map((shareholder) => {
+        if (
+          addresses.includes(shareholder.holder_id) &&
+          (shareholder.shares / total) * 100 >= min_shares
+        ) {
+          return contract.tokens
+        }
+        return undefined
+      }),
+      (token) => token !== undefined
```

Comment:
isn't this `tokens`? (plural?)

Reply:
You mean the last one? It's to filter undefined token (singular) in the tokens array

Reply:
yes, the last one. hmm, in the map function above you are returning `contract.tokens` (plural).

Reply:
at least to me, it looks like the `const tokens` from line 261 contains an array of arrays.

Reply:
True there might be better, but I'm deconstructing it line 274

Reply:
Or are you just referring to renaming the variable token to token_list for instance line 271? 🤣

Reply:
yes, it's technically correct. just the variable name `token` could be changed to `tokens`

Reply:
Ahahaha! Will do those edits